authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/src/main/java/org/wso2/carbon/identity/sample/oauth2/federated/authenticator/OAuth2CustomAuthenticator.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ ******************************************************************************/
+
+package org.wso2.carbon.identity.sample.oauth2.federated.authenticator;
+
+import org.wso2.carbon.identity.application.authenticator.oauth2.Oauth2GenericAuthenticator;
+import org.wso2.carbon.identity.application.common.model.Property;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.sample.oauth2.federated.authenticator.OAuth2CustomAuthenticatorConstants.*;
+
+/***
+ * OAuth2 Custom Authenticator is an outbound authenticator
+ * This extends OAuth2 Generic Authenticator implementation
+ */
+public class OAuth2CustomAuthenticator extends Oauth2GenericAuthenticator {
+
+    private static final long serialVersionUID = 6614257960044886319L;
+
+    @Override
+    public String getFriendlyName() {
+
+        return AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getName() {
+
+        return AUTHENTICATOR_NAME;
+    }
+
+    // TODO: Override buildClaims to match the response of respective OAuth2 identity provider
```

Comment:
Why do we have a TODO committed? Have we addressed this case later in a PR?

Reply:
This is to notify the developer to override the buildClaims method and do the implementation as necessary.